app/forms/merits_assessments/client_received_legal_help_form.rb

```diff
@@ -0,0 +1,19 @@
+module MeritsAssessments
+  class ClientReceivedLegalHelpForm
+    include BaseForm
+
+    form_for MeritsAssessment
+
+    attr_accessor :client_received_legal_help, :application_purpose
+
+    before_validation :clear_application_purpose
+    validates :client_received_legal_help, presence: true
+    validates :application_purpose, presence: true, if: proc { |form| form.client_received_legal_help.to_s == 'false' }
+
+    private
+
+    def clear_application_purpose
+      application_purpose.clear if client_received_legal_help.to_s == 'true'
+    end
```

Comment:
I wonder if this would be a little cleaner:
```ruby
validates :application_purpose, presence: true, unless: :client_received_legal_help?

private

def client_received_legal_help?
  client_received_legal_help.to_s == 'true'
end

def clear_application_purpose
  application_purpose.clear if client_received_legal_help?
end
```
It means that both the clearing process and the validation use the same comparison logic and handles instances where `client_received_legal_help` is neither 'true' nor 'false'.

Reply:
The code is much cleaner but it doesn't work perfectly.
We want to do `validates :application_purpose, presence: true` only client_received_legal_help has been selected to "No".
So if the user didn't select anything, we don't want to trigger that validation.

i.e.:
We don't want to show the second error message
![image](https://user-images.githubusercontent.com/482806/51678773-21f96b80-1fd5-11e9-889d-cfb1f0e1f940.png)

Reply:
:+1: Fair enough.